### PR TITLE
Fixed last login not updating issue.

### PR DESCRIPTION
**Pull request by Buwujiu**

### CODE REVIEW

**Comment by Buwujiu** [comment]
Two changes have been made for this issue:
1. TokenService: userEvent.update() is replaced by userService.update(), note this is only a workaround for updating lastlogin, the root cause is threading, another ticket will be opened to further investigation. 
2. A new test case is created in TokenServiceTest for testing lastlogin.

File: src/test/java/org/overture/ego/token/TokenServiceTest.java [hunk 1]
```diff
@@ -72,4 +76,20 @@ public void generateUserToken() {
...
+  @Test
+  public void testLastloginUpdate(){
+    IDToken idToken = new IDToken();
+    idToken.setFamily_name("fName");
+    idToken.setGiven_name("gName");
+    idToken.setEmail("fNamegName@domain.com");
+    User user = userService.create(entityGenerator.createOneUser(Pair.of("fName", "gName")));
+
+    assertNull(" Verify before generatedUserToken, last login should be null. ", user.getLastLogin());
+
```
[the remark this answers is not in the document]
Comment (Buwujiu) [reply src/test/java/org/overture/ego/token/TokenServiceTest.java]: test case updated!

File: src/test/java/org/overture/ego/token/TokenServiceTest.java [hunk 2]
```diff
@@ -55,6 +58,7 @@
   private TokenService tokenService;
 
   @Test
+  @Ignore
```
[the remark this answers is not in the document]
Comment (Buwujiu) [reply src/test/java/org/overture/ego/token/TokenServiceTest.java]: Ignored test case is removed.

File: src/main/java/org/overture/ego/token/TokenService.java [hunk 3]
```diff
@@ -80,11 +80,8 @@ public String generateUserToken(IDToken idToken){
       }
     }
 
-    // Update user.lastLogin in the DB
-    // Use events as these are async:
-    //    the DB call won't block returning the Token
     user.setLastLogin(new Date());
-    userEvents.update(user);
+    userService.update(user);
```
[the remark this answers is not in the document]
Comment (Buwujiu) [reply src/main/java/org/overture/ego/token/TokenService.java]: After the discussion, we decided to keep using userService as a workaround, another ticket will be created for further investigation.

File: src/main/java/bio/overture/ego/model/entity/User.java [hunk 4]
```diff
@@ -278,14 +279,17 @@ public void update(User other) {
     // To clear wholeApplications, wholeGroups or userPermissions, use the dedicated services
     // for deleting associations or pass in an empty Set.
     if (other.wholeApplications != null) {
+      SessionUtils.unsetSession(other.getWholeApplications());
```
Comment (rtisma) [inline src/main/java/bio/overture/ego/model/entity/User.java]: statically import this please. it improves readability

**Comment by rtisma** [comment]
@Buwujiu  great work!

**Pull request merged**